tests/integration/test_tracking.py

```diff
@@ -1136,6 +1137,7 @@ def test_memory_records_tick_interval(self, tmp_path):
 
         assert len(memory_records)
         assert all(record.rss > 0 for record in memory_records)
+        assert all(record.heap > 0 for record in memory_records)
```

Comment:
As above, seems racy

Reply:
I changed this to check if a heap field equals the heap size at some point, which is less racy and probably more aligned to what we want to check here.